src/java/fr/paris/lutece/plugins/document/business/DocumentDAO.java

```diff
@@ -182,29 +183,31 @@ public int newPrimaryKey( )
      */
     public synchronized void insert( Document document )
     {
-        DAOUtil daoUtil = new DAOUtil( SQL_QUERY_INSERT );
-        daoUtil.setInt( 1, document.getId( ) );
-        daoUtil.setString( 2, document.getCodeDocumentType( ) );
-        daoUtil.setString( 3, document.getTitle( ) );
-        daoUtil.setTimestamp( 4, document.getDateCreation( ) );
-        daoUtil.setTimestamp( 5, document.getDateModification( ) );
-        daoUtil.setString( 6, document.getXmlWorkingContent( ) );
-        daoUtil.setString( 7, document.getXmlValidatedContent( ) );
-        daoUtil.setInt( 8, document.getSpaceId( ) );
-        daoUtil.setInt( 9, document.getStateId( ) );
-        daoUtil.setString( 10, document.getSummary( ) );
-        daoUtil.setString( 11, document.getComment( ) );
-        daoUtil.setTimestamp( 12, document.getDateValidityBegin( ) );
-        daoUtil.setTimestamp( 13, document.getDateValidityEnd( ) );
-        daoUtil.setString( 14, document.getXmlMetadata( ) );
-        daoUtil.setInt( 15, document.getCreatorId( ) );
-        daoUtil.setInt( 16, document.getMailingListId( ) );
-        daoUtil.setInt( 17, document.getPageTemplateDocumentId( ) );
-        daoUtil.setBoolean( 18, document.isSkipPortlet( ) );
-        daoUtil.setBoolean( 19, document.isSkipCategories( ) );
-
-        daoUtil.executeUpdate( );
-        daoUtil.free( );
+        try ( DAOUtil daoUtil = new DAOUtil( SQL_QUERY_INSERT ) )
+        {
+            daoUtil.setInt( 1, document.getId( ) );
+            daoUtil.setString( 2, document.getCodeDocumentType( ) );
+            daoUtil.setString( 3, document.getTitle( ) );
+            daoUtil.setTimestamp( 4, document.getDateCreation( ) );
+            daoUtil.setTimestamp( 5, document.getDateModification( ) );
+            daoUtil.setString( 6, document.getXmlWorkingContent( ) );
+            daoUtil.setString( 7, document.getXmlValidatedContent( ) );
+            daoUtil.setInt( 8, document.getSpaceId( ) );
+            daoUtil.setInt( 9, document.getStateId( ) );
+            daoUtil.setString( 10, document.getSummary( ) );
+            daoUtil.setString( 11, document.getComment( ) );
+            daoUtil.setTimestamp( 12, document.getDateValidityBegin( ) );
+            daoUtil.setTimestamp( 13, document.getDateValidityEnd( ) );
+            daoUtil.setString( 14, document.getXmlMetadata( ) );
+            daoUtil.setInt( 15, document.getCreatorId( ) );
+            daoUtil.setInt( 16, document.getMailingListId( ) );
+            daoUtil.setInt( 17, document.getPageTemplateDocumentId( ) );
+            daoUtil.setBoolean( 18, document.isSkipPortlet( ) );
+            daoUtil.setBoolean( 19, document.isSkipCategories( ) );
+
+            daoUtil.executeUpdate( );
+            daoUtil.free( );

```

Comment:
"daoUtil.free( );"  is no longer necessary in a "try with resource" block, 
please remove this and all occurrences above